projects/storefrontlib/src/cms-components/myaccount/my-interests/my-interests.component.spec.ts

```diff
@@ -86,180 +88,137 @@ const mockedInterests: ProductInterestList = {
+  it('should show interests list', () => {
+    userService.getProdutInterests.and.returnValue(of(mockedInterests));
+    userService.getProdutInterestsLoaded.and.returnValue(of(true));
     fixture.detectChanges();
 
-    expect(el.query(By.css('.cx-product-interests-no-interests'))).toBeTruthy();
+    expect(el.queryAll(By.css('cx-sorting')).length).toEqual(2);
+    expect(el.queryAll(By.css('cx-pagination')).length).toEqual(2);
+    expect(el.queryAll(By.css('tr')).length).toEqual(2);
   });
 
-  it('should display interests list correctly', () => {
-    spyOn(userService, 'getProdutInterests').and.returnValue(
-      of(mockedInterests)
-    );
-    component.ngOnInit();
+  it('should be able to change page/sort', () => {
     fixture.detectChanges();
-    expect(el.query(By.css('.cx-product-interests-table'))).toBeTruthy();
-  });
 
-  it('should be able to click image and product hyperlink', () => {
-    spyOn(userService, 'getProdutInterests').and.returnValue(
-      of(mockedInterests)
-    );
-    component.ngOnInit();
-    fixture.detectChanges();
+    component.sortChange('sort');
+    expect(userService.getProdutInterests).toHaveBeenCalled();
 
-    const tr = el.query(By.css('.cx-product-interests-table tbody tr'));
-    expect(
-      tr.query(By.css('.cx-product-interests-product-image-link'))
-    ).toBeTruthy();
-    expect(
-      tr.query(By.css('.cx-product-interests-product-code-link'))
-    ).toBeTruthy();
+    component.pageChange(2);
+    expect(userService.getProdutInterests).toHaveBeenCalled();
   });
 
   it('should be able to delete an interest item', () => {
-    const interests: ProductInterestList = { ...mockedInterests };
-    spyOn(userService, 'getProdutInterests').and.returnValue(of(interests));
-    spyOn(userService, 'deleteProdutInterest').and.callFake((_item: any) => {
-      interests.results = null;
-      interests.pagination.totalCount = 0;
-    });
-    component.ngOnInit();
+    userService.getProdutInterests.and.returnValue(of(mockedInterests));
+    userService.deleteProdutInterest.and.stub();
     fixture.detectChanges();
-    el.query(By.css('.close')).nativeElement.click();
+    el.query(By.css('button')).nativeElement.click();
     fixture.detectChanges();
 
-    expect(el.query(By.css('.cx-product-interests-no-interests'))).toBeTruthy();
+    expect(userService.deleteProdutInterest).toHaveBeenCalled();
   });
 
-  it('should not display stock status when in stock', () => {
-    const interests: ProductInterestList = { ...mockedInterests };
-    interests.results[0].product.stock.stockLevel = 10;
-    interests.results[0].product.stock.stockLevelStatus = 'inStock';
-    spyOn(userService, 'getProdutInterests').and.returnValue(of(interests));
-    component.ngOnInit();
-    fixture.detectChanges();
-
-    expect(el.query(By.css('.cx-product-interests-product-stock'))).toBeFalsy();
+  it('should reset value when the component is destroy', () => {
+    userService.clearProductInterests.and.stub();
```

Comment:
is it needed or not (clearProductInterests is used for ??)

Reply:
no need, remove it.